packages/react-utilities/src/compose/getSlots.ts

```diff
@@ -1,8 +1,87 @@
 import * as React from 'react';
 
-import { getNativeElementProps, omit } from '../utils/index';
-import { GenericDictionary } from './types';
+import { ComponentState, ShorthandRenderFunction, SlotPropsRecord } from './types';
 import { nullRender } from './nullRender';
+import { getNativeElementProps, omit } from '../utils/index';
+
+function getSlot(
+  component: React.ElementType | undefined,
+  alternative: keyof JSX.IntrinsicElements | undefined,
+  defaultValue: React.ElementType,
+) {
+  return (
+    (component === undefined || typeof component === 'string' ? alternative ?? component : component) ?? defaultValue
```

Comment:
nit It's a matter of preference but I suggest to simplify conditions there to multiple `if`s to make it more readable